src/Dapr.Jobs/ScheduleJobOptions.cs

```diff
@@ -0,0 +1,31 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2024 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using System.Text.Json.Serialization;
+
+namespace Dapr.Jobs;
+
+/// <summary>
+/// The options used to schedule a new job.
+/// </summary>
+/// <param name="JobData">The internal job options.</param>
+internal record ScheduleJobOptions([property: JsonPropertyName("job")] ScheduleJobInnerOptions JobData);
+
+/// <summary>
+/// The payload used to schedule a new job.
+/// </summary>
+/// <param name="JsonSerializableData">The JSON serializable data payload.</param>
+/// <param name="DueTime">The </param>
+internal record ScheduleJobInnerOptions(
+    [property: JsonPropertyName("data")] object JsonSerializableData, 
+    [property: JsonPropertyName("dueTime")] string DueTime);
```

Comment:
Ideally the `DueTime` should be a `TimeSpan` (and converted into whatever format is used by Dapr).  There are also additional properties (such as recurrence) which appear to be supported in the gRPC API but not mentioneed in the HTTP-based Dapr preview docs.  We'll want to incorporate those as well.

Reply:
The protos still appear to use a `string` instead of the common gRPC `Duration` type, which I find odd. In any case, we should attempt to *not* require users of this .NET API to understand curiosities of the Golang runtime.

Reply:
There may be enough parameters on the gRPC interface to justify an "options" argument type rather than try to have them all as individual arguments.

Reply:
I was thinking of just using a mix of overloads as jobs can be scheduled with either a Cron string or a TimeSpan. I've got the extension method to handle their golang TimeSpan expression already done and tested, but by using arguments, we could more easily guide users to using one or the other and avoiding a situation where they populate both and.. probably something undesirable happens.

Reply:
Let's see how that ends up.  Another option would be to abstract the notion of a duration into a special type (e.g. `Duration`) and then have converters for the various formats (e.g. `Duration.FromTimeSpan(TimeSpan ts)` and `Duration.FromCronString(string cs)`).

Reply:
This type has been removed and we've got the discussion continuing below about this, so marking this part as resolved for now.